aws-xray-auto-instrumentation-agent-runtime/src/main/java/com/amazonaws/xray/agent/config/AgentConfiguration.java

```diff
@@ -0,0 +1,150 @@
+package com.amazonaws.xray.agent.config;
+
+import java.util.Objects;
+
+/**
+ * Class represents the agent configuration's JSON file
+ */
+public class AgentConfiguration {
```

Comment:
Sorry I didn't get to take a deep look at this PR but one high level comment is I would recommend making these configurations immutable. Immutable objects tend to be less fragile because callers know values won't change on them (for example a value can be stored into a callback if needed). And they're guaranteed threadsafe instead of hoping it doesn't get mutated at the right time.

Do you think we can rework the code so the configurations are immutable?

Reply:
As a potentially reference here is the OTel agent config class, just to get an idea. It's huge and uses lombok, which we shouldn't use :) But might be an interesting read.

Reply:
I was reading into the OT config class and got some pointers from it, definitely a good point of reference. That's a very good point about immutability, the refactor shouldn't be too tough.

Reply:
Updated! This also made the code simpler by ensuring most fields of the configuration are assigned to valid values by default, which removed some of the validation we had to do. It also simplified the parsing logic and moved most of it into the `AgentConfiguration` class, which is cleaner IMO.